File: src/lib/types/prestataires.ts
// Types prestataires — Phase 5
export interface Prestataire {
  id_prestataire: number;
  libelle: string;
  description: string | null;
  adresse: string | null;
  code_postal: string | null;
  ville: string | null;
  telephone: string | null;
  email: string | null;
  id_image: number | null;
}

File: src-tauri/src/models/prestataires.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Prestataire {
    pub id_prestataire: i64,
    pub libelle: String,
    pub description: Option<String>,
    pub adresse: Option<String>,
    pub code_postal: Option<String>,
    pub ville: Option<String>,
    pub telephone: Option<String>,
    pub email: Option<String>,
    pub id_image: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct PrestataireInput {
    pub libelle: String,
    pub description: Option<String>,
    pub adresse: Option<String>,
    pub code_postal: Option<String>,
    pub ville: Option<String>,
    pub telephone: Option<String>,
    pub email: Option<String>,
    pub id_image: Option<i64>,
}
